### PR TITLE
[BCF-3200] Make sure short tests are actually passing

### PR DESCRIPTION
**Pull request by HenryNguyen5**

### CODE REVIEW

**Review by HenryNguyen5 — COMMENTED** [review]
Just realized `detect-flakey-tests:` is almost a full copy/paste of our regular tests... I forgot to update the database URL in there. @kalverra why don't we build the flakey test detection into the ci core test itself?

File: README.md [hunk 1]
```diff
@@ -199,7 +199,9 @@ go test ./...
 
 - The `parallel` flag can be used to limit CPU usage, for running tests in the background (`-parallel=4`) - the default is `GOMAXPROCS`
 - The `p` flag can be used to limit the number of _packages_ tested concurrently, if they are interferring with one another (`-p=1`)
-- The `-short` flag skips tests which depend on the database, for quickly spot checking simpler tests in around one minute
+- The `-short` flag skips tests which depend on the database, for quickly spot checking simpler tests in around one minute. We have some validation tests that still need a database url to verify configuration, although they don't actually use the database. You should pass in a fake url to run short tests. `CL_DATABASE_URL='postgresql://doesnotexist:justtopassvalidation@localhost:5432/chainlink_na_test?sslmode=disable' go test -short ./...`
```
Comment (jmank88) [inline README.md:202]: Which tests still use the DB?

Comment (HenryNguyen5) [reply README.md:202]: It's confusing. The tests don't need a DB, they need a DB url. It's tests like `core/chain/evm/config` which check the DB url to see if it has sufficient password complexity. They fail without the env var set.

Comment (jmank88) [reply README.md:202]: Hmm that sounds odd. It might just be appropriate to skip them

File: README.md [hunk 2]
```diff
@@ -199,7 +199,9 @@ go test ./...
 
 - The `parallel` flag can be used to limit CPU usage, for running tests in the background (`-parallel=4`) - the default is `GOMAXPROCS`
 - The `p` flag can be used to limit the number of _packages_ tested concurrently, if they are interferring with one another (`-p=1`)
-- The `-short` flag skips tests which depend on the database, for quickly spot checking simpler tests in around one minute
+- The `-short` flag skips tests which depend on the database, for quickly spot checking simpler tests in around one minute.
+
+```bash
```
Comment (jmank88) [inline README.md:204]:
```suggestion
```

**Review by jmank88 — CHANGES_REQUESTED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**